### PR TITLE
refactor(ruleset): Tweak to the ScheduleRuleset schema

### PR DESCRIPTION
**Pull request by chriswmackey**
After realizing that the current ScheduleRuleset schema is not very good at supporting serialization from IDF and OSM formats, I have made a slight tweak to it.  The schedule_primary_school_occupancy.json file in this PR probably shows this change the best.  Essentially, I realized it's very useful to have the ScheduleDays under their own tag within the ScheduleRuleset and have all the rules and other schedule reference these ScheduleDays by name.  It helps a lot in managing cross-referenced ScheduleDays from imported IDFs and it also helps slim down the text to represent ScheduleRules that change over smaller time intervals (like every month).

More reasons for the change can be found in [this PR](https://github.com/ladybug-tools/honeybee-energy/pull/57) on honeybee-energy.

### CODE REVIEW

File: app/models/samples/json/model_with_humidity_setpoints.json [hunk 1]
```diff
@@ -413,6 +413,88 @@
                 }
             ],
             "schedules": [
+                {
+                    "type": "ScheduleRulesetAbridged",
+                    "name": "Hospital INFIL_SCH_PNNL",
+                    "default_day_schedule": {
```
Comment (tanushree04) [inline app/models/samples/json/model_with_humidity_setpoints.json]: @chriswmackey , this example file is missing the field day_schedules and does not have referencing for schedule day in default day schedule, summer and winter design day schedule

File: app/models/samples/json/schedule_ruleset_office_occupancy.json [hunk 2]
```diff
@@ -118,7 +118,7 @@
             "type": "ScheduleRule",
             "schedule_day": {
                 "type": "ScheduleDay",
-                "name": "Large Office Bldg Occ_WntrDsnSun",
```
Comment (tanushree04) [inline app/models/samples/json/schedule_ruleset_office_occupancy.json:121]: @chriswmackey , this example file does not include the new changes to scheduleruleset as well

**Comment by chriswmackey** [comment]
Thank you for catching those issues @tanushree04 .  I realized that not all of the code updates I made to the Python libraries  have been coordinated on my end.  I'll add some corrected versions shortly.

**Comment by chriswmackey** [comment]
The issues with those two files should be addressed now.  Feel free to merge whenever it's most convenient.

**Comment by tanushree04** [comment]
@chriswmackey It is still failing CI. Should I merge in my branch? That includes the updated examples as well.

**Comment by chriswmackey** [comment]
@tanushree04 ,

The reason why the tests are failing is because I only updated the JSON samples and not any of the pydantic files or validation.  I believe that the tests should pass when we combine this PR with the one you have been working on. I'll merge this in now so that you can pull the changes into the branch from which you are making your PR.

**Pull request merged**